### DISCUSSION TITLE
How I can add malformed row to Datatable if CsvHelperLoadError raised

### DISCUSSION THREAD

**Discussion by ViacheslavUKR** [JoshClose/CsvHelper#1363] [Q&A]
Hello, I read CSV data by the pattern below. Unfortunately I have a lot of malformed rows (they contains more fields than header). How I can processed they by own code and add to Datatable part of malformed rows or marker of bad data? Thanks.

    Function ReadCSV(CsvFileName As String) As DataTable
        'https://joshclose.github.io/CsvHelper/examples/reading/
        ErrRowListNumber = New List(Of Integer)
        Dim RDR = New StreamReader(CsvFileName)
        Dim CSV = New CsvHelper.CsvReader(RDR)
        CSV.Configuration.BadDataFound = (Sub(X) MsgBox("Import this row is impossible (incorrect delimiter):" & vbCrLf & X.RawRecord))
        CSV.Configuration.Delimiter = ";"
        Dim DR = New CsvHelper.CsvDataReader(CSV)
        Dim DT As New DataTable()
        DT.Load(DR, LoadOption.OverwriteChanges, AddressOf CsvHelperLoadError)
        RDR.Close()
        RDR.Dispose()
        CSV = Nothing
        DR.Close()
        DR.Dispose()
        DR = Nothing
        Return DT
    End Function

    Dim ErrRowListNumber As List(Of Integer)
    Sub CsvHelperLoadError(sender As Object, e As FillErrorEventArgs)
        e.DataTable.Rows.Add()
        Dim Str1 As New Text.StringBuilder
        For i As Integer = 0 To e.Values.Count - 1
            'e.DataTable.Rows(e.DataTable.Rows.Count - 1)(i) = e.Values(i) - impossible, readonly
            Str1.AppendLine("(" & i & ")" & e.Values(i))
        Next
        MsgBox("Import this row is impossible (more than " & e.Values.Count & " column):" & vbCrLf & Str1.ToString)
        ErrRowListNumber.Add(e.DataTable.Rows.Count)
        e.Continue = True
    End Sub

**Comment by A9G-Data-Droid** [comment]
Your question is about `DataTable.Load` and not this library at all. 

You should go to StackOverflow for answers: https://stackoverflow.com/questions/41758708/how-do-i-handle-errors-with-dataset-fill

The answer is to use `.Add` to add a row with something during the error.
